go/channelling/roomworker.go

```diff
@@ -290,6 +271,11 @@ func (r *roomWorker) Leave(sessionID string) {
 		r.mutex.Lock()
 		if _, ok := r.users[sessionID]; ok {
 			delete(r.users, sessionID)
+			if len(r.users) == 0 {
+				r.mutex.Unlock()
+				r.expired <- true
```

Comment:
I think only this change is required to immediately wake the room worker when the last user leaves. The other changes here can imho be reverted.

But please change the channel send to be nonblocking (and move to a method to also call from the timer)
```
func (r *roomWorker) wakeup() {
    select {
    case r.expired <- true:
    default:
    }
}
```